src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -268,20 +282,85 @@ public AcceptTxResult SubmitTx(Transaction tx, TxHandlingOptions handlingOptions
                     $"Adding transaction {tx.ToString("  ")} - managed nonce: {managedNonce} | persistent broadcast {startBroadcast}");
 
             TxFilteringState state = new(tx, _accounts);
-            for (int i = 0; i < _filterPipeline.Count; i++)
+
+            AcceptTxResult accepted = FilterTransactions(tx, handlingOptions, state);
+
+            if (!accepted)
             {
-                IIncomingTxFilter incomingTxFilter = _filterPipeline[i];
-                AcceptTxResult accepted = incomingTxFilter.Accept(tx, state, handlingOptions);
-                if (!accepted)
-                {
-                    Metrics.PendingTransactionsDiscarded++;
-                    return accepted;
-                }
+                Metrics.PendingTransactionsDiscarded++;
+                return accepted;
             }
 
             return AddCore(tx, startBroadcast);
         }
 
+        private AcceptTxResult FilterTransactions(Transaction tx, TxHandlingOptions handlingOptions, TxFilteringState state)
```

Comment:
Is it worth it compared to a clean pipeline? I would say the filters have such a big impact in their logic that this is micro-optimization that doesn't make a dent in the performance, but make the code uglier.

Reply:
Changed